ably/ably-chat-js
# react-hooks: Add usePresence hook

### Context

* [CHA-387]
* As part of the react hooks development, a new hook to interact with the `Presence` instance of a room was required.

### Description

- Added the `usePresence` hook to the Chat React library. This allows the user to interact with the underlying `Presence` instance, as well as easily update and leave presence.
- A user is entered into presence on rendering the hook.
- Updated the demo app to use the new hook, this includes updating and leaving presence.
  - Entering presence is now automatic on calling the hook.
- Updated the readme to document hook usage.
- Added unit tests for the new hook.


### Checklist


### Testing Instructions (Optional)

1. Start a local demo app
2. On loading the app, you should see your clientId added to the presence list.
3. Check opening multiple tabs adds the users to the presence list.
4. Check unmounting the component leaves presence and the presence lists of other clients are updated.
5. Check the leave button leaves presence.
6. Check the update button sets the presence data.


[CHA-387]: https://ably.atlassian.net/browse/CHA-387?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
